ckb-contracts/lib/eth-recipient-typescript-lib/src/adapter/mod.rs

```diff
@@ -1,17 +1,134 @@
 #[cfg(target_arch = "riscv64")]
 pub mod chain;
+use crate::actions::calc_eth_bridge_lock_hash;
+
+#[cfg(not(feature = "std"))]
+use alloc::{vec, vec::Vec};
+
+use ckb_std::ckb_constants::Source;
+use ckb_std::ckb_types::{
+    bytes::Bytes,
+    packed::{Byte32, Script},
+    prelude::Pack,
+};
+use ckb_std::error::SysError;
+use core::convert::TryFrom;
+use force_eth_types::config::{SUDT_CODE_HASH, SUDT_HASH_TYPE, UDT_LEN};
+use force_eth_types::eth_recipient_cell::{ETHAddress, ETHRecipientDataView};
 #[cfg(feature = "std")]
 use mockall::predicate::*;
 #[cfg(feature = "std")]
 use mockall::*;
-
-use ckb_std::ckb_constants::Source;
-
-use force_eth_types::eth_recipient_cell::ETHRecipientDataView;
+use molecule::prelude::{Builder, Byte, Entity};
 
 #[cfg_attr(feature = "std", automock)]
 pub trait Adapter {
-    fn load_output_data(&self) -> Option<ETHRecipientDataView>;
+    fn load_output_data_by_trait(&self) -> Vec<Vec<u8>>;
+
+    fn load_cell_data_by_trait(&self, index: usize, source: Source) -> Vec<u8>;
+
+    fn load_cell_type_by_trait(
+        &self,
+        index: usize,
+        source: Source,
+    ) -> Result<Option<Script>, SysError>;
+}
+
+pub fn load_output_data(adapter: &dyn Adapter) -> Option<ETHRecipientDataView> {
```

Comment:
It's better to use trait function with default implementation so we can test it easily by mock.